torchgeo/trainers/cowc.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""COWC dataset trainer."""
+
+from typing import Any, Dict, Optional
+
+import pytorch_lightning as pl
+from torch.utils.data import DataLoader, random_split
+
+from ..datasets import COWCCounting
+
+# https://github.com/pytorch/pytorch/issues/60979
+# https://github.com/pytorch/pytorch/pull/61045
+DataLoader.__module__ = "torch.utils.data"
+
+
+class COWCCountingDataModule(pl.LightningDataModule):
+    """LightningDataModule implementation for the COWC Counting dataset."""
+
+    def __init__(
+        self,
+        root_dir: str,
+        seed: int,
+        batch_size: int = 64,
+        num_workers: int = 4,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a LightningDataModule for COWC Counting based DataLoaders.
+
+        Args:
+            root_dir: The ``root`` arugment to pass to the
+                COWCCounting Datasets classes
+            seed: The seed value to use when doing the sklearn based GroupShuffleSplit
+            batch_size: The batch size to use in all created DataLoaders
+            num_workers: The number of workers to use in all created DataLoaders
+            api_key: The RadiantEarth MLHub API key to use if the dataset needs to be
+                downloaded
+        """
+        super().__init__()  # type: ignore[no-untyped-call]
+        self.root_dir = root_dir
+        self.seed = seed
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+
+    def custom_transform(self, sample: Dict[str, Any]) -> Dict[str, Any]:
+        """Transform a single sample from the Dataset.
+
+        Args:
+            sample: dictionary containing image and target
+
+        Returns:
+            preprocessed sample
+        """
+        sample["image"] = sample["image"] / 255.0  # scale to [0,1]
+        sample["label"] = sample["label"].float()
```

Comment:
Is this key correct for regression datasets?

Reply:
Yes, all scalar values use "label" and all matrices use "mask".

Reply:
See #213 for more discussion

Reply:
Ah nice